covalent_ui/webapp/src/components/settings/SettingsCard.js

```diff
@@ -0,0 +1,1062 @@
+  const [handle, setHandle] = useState('')
+  const [searchData, setSearchData] = useState(settings_result)
+  const [searchKey, setSearchKey] = useState('')
+  const [restoreData, setRestoreData] = useState()
+  const [valueChange, setValueChange] = useState(false)
+  const [clientDetail, setClientDetail] = useState(null)
+  const [serverDetail, setServerDetail] = useState(null)
+  const [tempData, setTempData] = useState(null)
+  const [tempDataServer, setTempDataServer] = useState(null)
+
+  useEffect(() => {
+    dispatch(settingsResults())
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (valueChange) {
+      const settingObj = {
+        isChanged: valueChange,
+        data: resultOutput,
+        nodeName: resultKey,
+        mainKey: accName,
+      }
+      dispatch(toggleLatticeDrawer(settingObj))
+    } else {
+      const settingObj = {
+        isChanged: valueChange,
+      }
+      dispatch(toggleLatticeDrawer(settingObj))
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [valueChange, resultOutput])
+
+  useEffect(() => {
+    if (settings_result) {
+      setSearchData(Object.values(settings_result)[0])
+      setClientDetail(Object.values(settings_result)[0])
+      setServerDetail(Object.values(settings_result)[1])
+      setTempData(Object.values(settings_result)[0])
+      setTempDataServer(Object.values(settings_result)[1])
+    }
+  }, [settings_result])
+
+  useEffect(() => {
+    if (_.size(searchData) !== 0) {
+      setResultOutput(Object.values(searchData)[0])
+      setRestoreData(Object.values(searchData)[0])
+    }
+  }, [searchData])
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    const updateData = {
+      [accName]: {
+        [resultKey]: resultOutput,
+      },
+    }
+    dispatch(updateSettings(updateData)).then((action) => {
+      if (action.type === updateSettings.fulfilled.type) {
+        setOpenSnackbar(true)
+        setSnackbarMessage('Settings Updated Successfully')
+        setValueChange(false)
+        setClientDetail((currValue) => ({
+          ...currValue,
+          [resultKey]: resultOutput,
+        }))
+        setRestoreData(resultOutput)
+      } else if (action.type === updateSettings.rejected.type) {
+        setOpenSnackbar(true)
+        setSnackbarMessage(
+          'Something went wrong and could not settings updated!'
+        )
+      }
+    })
+    setHandle('')
+  }
+
+  const getSubmenuName = (name) => {
+    let formattedName = name
+    const uSpliit = name.includes('_')
+    if (uSpliit) {
+      var a = name
+        .replace(name.at(0), name.at(0).toLocaleUpperCase())
+        .substring(0, name.indexOf('_'))
+      var b = name
+        .replace(name.at(0), name.at(0).toLocaleUpperCase())
+        .substring(name.indexOf('_') + 1, name.length)
+      formattedName = a + ' ' + b.replace(b.at(0), b.at(0).toLocaleUpperCase())
+    } else {
+      if (name === 'slurm') {
```

Comment:
For any string literals used let's create a seperate file that define these as constants grouped by their usage.

e.g

```
const CONFIG_SECTIONS = {
  SLURM: 'slurm'
}
```

such that this can be re-used as `CONFIG_SECTIONS.SLURM` and similarly for the other config sections